ddtrace/contrib/openai/_endpoint_hooks.py

```diff
@@ -3,25 +3,68 @@
 
 from ..trace_utils import set_flattened_tags
 from .utils import _est_tokens
+from .utils import _format_bool
+from .utils import _format_openai_api_key
 
 
-class _EndpointHook:
-    _request_tag_attrs = []
-    _default_name = "openai"
+API_VERSION = "v1"
+
 
-    def _record_request(self, span, kwargs):
-        for kw_attr in self._request_tag_attrs:
+class _EndpointHook:
+    # Assume base arg signature follows openai.EngineAPIResource.create(...)
+    _request_arg_params = ("cls", "api_key", "api_base", "api_type", "request_id", "api_version", "organization")
+    _request_kwarg_params = ()
+    _prompt_completion = False
+    ENDPOINT_NAME = "openai"
+    REQUEST_TYPE = ""
+    OPERATION_ID = ""
+
+    def _record_request(self, span, integration, args, kwargs):
+        """Set base-level openai tags, as well as request params from args and kwargs."""
+        endpoint = self.ENDPOINT_NAME
+        if endpoint is None:
+            endpoint = "%s" % args[0].OBJECT_NAME
+        span.set_tag_str("openai.request.endpoint", "/%s/%s" % (API_VERSION, endpoint))
+        span.set_tag_str("openai.request.method", self.REQUEST_TYPE)
+
+        base_level_tag_args = ("api_base", "api_type", "api_version")
+
+        if self._request_arg_params and len(self._request_arg_params) > 1:
+            for idx, arg in enumerate(self._request_arg_params[1:], 1):
+                if idx >= len(args):
+                    break
+                if arg is None or args[idx] is None:
+                    continue
+                if arg in base_level_tag_args:
+                    span.set_tag("openai.%s" % arg, args[idx] or "")
```

Comment:
This is intentional - this would only get tagged if the user specifically specified an empty value. The above line checks that the arg is not a None value.

Reply:
On second thought, we don't need this empty value default as this would be tagged if the value is not empty.